qchem/pennylane_qchem/qchem.py

```diff
@@ -639,7 +639,51 @@ def sd_excitations(n_electrons, n_orbitals, delta_sz=0):
              if (sz[p] + sz[q] - sz[r] - sz[s]) == delta_sz
     ]
 
-    return ph, pphh    
+    return ph, pphh
+
+
+def hf_state(n_electrons, m_spin_orbitals):
+    r"""Generates the occupation-number vector representing the Hartree-Fock (HF)
+    state of :math:`N` electrons in a basis of :math:`M` spin orbitals.
+
+    The many-particle wave function in the HF approximation is a `Slater determinant
+    <https://en.wikipedia.org/wiki/Slater_determinant>`_. In Fock space, a Slater determinant
+    is represented by the occupation-number vector:
+
+    .. math:
+        \vert {\bf n} \rangle = \vert n_1, n_2, \dots, n_M \rangle,
+
+        n_i = \left\lbrace \begin{array}{ll} 1 & i \leq N \\ 0 & i > N \end{array} \right..
+
+    **Example**
+
+    >>> init_state = hf_state(2, 6)
+    >>> print(init_state)
+    [1 1 0 0 0 0]
+
+    Args:
+        n_electrons (int): number of active electrons
+        m_spin_orbitals (int): number of active **spin-orbitals**
+
+    Returns:
+        array: NumPy array containing the vector :math:`\vert {\bf n} \rangle`
+    """
+
+    if not n_electrons > 0:
+        raise ValueError(
+            "The number of active electrons has to be > 0; got 'n_electrons' = {}"
+            .format(n_electrons)
+        )
+
+    if n_electrons > m_spin_orbitals:
+        raise ValueError(
+            "The number of active orbitals has to be >= the number of active electrons;"
+            " got 'm_spin_orbitals'={} < 'n_electrons'={}".format(m_spin_orbitals, n_electrons)
+        )        
+
+    hf_state_on = [1 if i < n_electrons else 0 for i in range(m_spin_orbitals)]
```

Comment:
Nitpick: Maybe this can be made more readable with NumPy?
```suggestion
    hf_state_on = np.where(np.arange(m_spin_orbitals) < n_electrons, 1, 0)
```

Reply:
(Since it's being converted to an array anyway)

Reply:
> (Since it's being converted to an array anyway)

nice!